### ISSUE TITLE
Homebridge/InsteonLocal Losing Connection to Insteon Hub

### ISSUE THREAD

**Issue by ericboles** [kuestess/homebridge-platform-insteonlocal#100]
My homebridge/Insteon setup had been working pretty reliably up until a few months ago, but it currently requires a daily reboot of the Insteon hub. 

For about a day after rebooting the Insteon hub, everything works normally...and then the Insteon plugin reports in homebridge an hourly loop of:
`Closing connection to the Hub...
Reconnecting to the Hub...`
Without ever reconnecting.  A reboot of the hub gets things working again, but only for about a day. 

Wondering what could be causing this.  Seems like it could be a faulty hub (although it’s running the latest firmware and works fine without a reboot with the native Insteon app). Either that or I’ve got something wonky with my config.json settings which I’ve tried to keep basic. They’re included below. 

Appreciate any insight. 

`     
"platforms": [
        {
            "platform": "InsteonLocal",
            "name": "Insteon Local Platform",
            "user": "theuser",
            "pass": "thepassword",
            "host": "10.0.1.42",
            "port": "25105",
            "model": "2245",
            "devices": [device list]
} `

**Comment by THX723** [comment]
Hmmm. I have noticed the same with my 2245 hub lately. I've had to default back to 'keep alive' every hour, but the random time-out persists. I'm also wondering if something was changed from firmware 1019 in early April.

**Comment by kuestess** [comment]
@ericboles @THX723 I had this issue too with my 2245 - no matter what I did it would randomly lose connection and not reconnect until a hard reboot.  Unfortunately, it ended with my Hub dying a few weeks later.  To their credit, Insteon replaced it outside of warranty.  In the meantime, maybe try disabling the keep alive and see if that helps.

**Comment by THX723** [comment]
@kuestess Same. My 2245 burned out and was very recently replaced by Insteon 'pro bono'. Awesome on Insteon!

I was running with 'keep alive' disabled, but still ran into the random disconnection. I'm going to explore tweaking network setting and see if that helps then report back. *Fingers crossed*

**Comment by ericboles** [comment]
@kuestess @THX723 Thanks for the insight.  I spent the weekend trying various settings in homebridge, but no luck getting it to maintain a connection more than a day.  It now looks like the Insteon Hub connection is starting to be unreliable through the Insteon app.  Sounds like a call to Insteon is in order to see about a replacement. Have either of you found that the replacement hub  fixed the issues?

**Comment by kuestess** [comment]
@ericboles Yep, totally fixed it.  They migrate your hub too, so all the devices are already linked and scenes remain intact.

**Comment by THX723** [comment]
@ericboles Yup, the new hub definitely fixed my hub crashing issues, which as you've pointed out will not even respond to the Insteon app. On the hardware side, it has been rock solid. No more red light, faint green light, or no light.

Having said that, Insteonlocal was still getting the random server no responses, but continue to respond to the Insteon app. :weird:

As a last resort, I tried disabling auto DHCP in the hub network setting and manually set it to the static IP. Before I was relying on my router to dynamically assign the static IP to the hub using DHCP.  It has been about a week and I have yet observer a single server no responses by Insteonlocal. Coincidence? Too early to say conclusively. We shall in see in time.

**Comment by ericboles** [comment]
@kuestess Good to hear a replacement may fix it...and even better that the hub is migrated.  Certainly saves some hassle.

@THX723 Good to keep in mind about the Hub DHCP.  I also have my router assign a static IP via DHCP--hasn't been a problem before, but could be a culprit if it's a lease issue that's causing the failed connections after ~24 hours...

**Comment by THX723** [comment]
Just providing a continual update.

Since making the the DHCP setting change, my 2445 hub has yet exhibit any no response conditions, while `keepAlive = 0` or disabled.

I wouldn't quite endorse this as the end-all yet, but it is certainly looking encouraging.

**Comment by ericboles** [comment]
A quick update from my end...  I went back and forth with Insteon support for a few weeks as my Hub was continually unresponsive after 24 hours but never fully "dead" (The light would mostly be green...but sometimes go to red...and then back to green...but still require a power cycle to get things back online every ~24 hours).  Tried disabling DHCP as @THX723 suggested, but still had the same issue.

Support was quite helpful, but their troubleshooting and fixes didn't solve the problem and they sent a new hub which arrived last week.  Happy to report that since installing it 3 days ago, everything has worked consistently.

My circumstantial conclusion is that it was a hardware rev/firmware upgrade issue.  Despite the same model number (2245-222), my old hub was hardware rev 1.4.  The new one is hardware rev 3.4.  In addition, my issues seemed to start in early April when the 1019 firmware update was pushed.  Time will tell whether the fix is truly permanent...but so far, so good.

**Comment by THX723** [comment]
@ericboles My old hub had the same issues until it was replaced by Insteon. However the disconnects persisted even with the new hub (v3.3), until I made the DHCP change.

**Comment by ericboles** [comment]
Thanks @THX723. I’ve had my UniFi system providing a static IP to the Insteon hub through DHCP for years without issue. Per your recommendation earlier, I did try disabling auto DHCP and manually setting the IP in the hub settings. Unfortunately that didn’t work with the old hub and I still lost connection to Homebridge after about a day. 

I had since changed it back to get the static IP through DHCP and that’s what the new hub is doing. Going on 5 days without issue (knock on wood). 

Certainly wish there was a clear common culprit...but it seems like a new hub fixed things for all 3 of us in different ways.
